openlibrary/plugins/openlibrary/home.py

```diff
@@ -77,17 +81,30 @@ def popular_carousel(limit=36):
         random.shuffle(seeds1)
         random.shuffle(seeds2)
         seeds = seeds1 + seeds2
+
+        parts = {}
         while seeds and len(books) < limit:
-            seed = seeds.pop(0)
-            key = seed['key']
-            ebook = seed.get_ebook_info()
-            if 'daisy_url' in ebook and 'borrow_url' not in ebook:
-                continue
-            if 'borrow_url' in ebook and ebook['borrowed']:
-                continue
-            edition = web.ctx.site.get(key)
-            book = format_book_data(edition)
-            books.append(book)
+            archive_ids = []
+            while seeds and len(parts) < 50:
+                seed = seeds.pop(0)
+                key = seed['key']
+                edition = web.ctx.site.get(key)
+                archive_id = edition.get('ocaid')
+                if archive_id:
+                    parts[archive_id] = edition
+                    archive_ids.append(archive_id)
+
+            responses = lending.is_borrowable(archive_ids)
+
+            for archive_id in archive_ids:
+                if len(books) == 36:
```

Comment:
There's no great reason for the hard-coded 50. The goal is to fill up 6 carousel pages of 6-book (i.e. 36 books). Since many books are checked out, we attempt to fetch batches of ~50 at a time to bulk check against the loans API.

We could load an arbitrary number of pages via ajax, but pagination becomes slightly complicated given we're presently randomizing our seeds.